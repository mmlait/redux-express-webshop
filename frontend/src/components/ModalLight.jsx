import React from 'react';
import styled from 'styled-components';
import colors from '../colors';

const ModalWrapper = styled.div`
  background-color: ${colors.modalBgOverlay};
  position: fixed;
  z-index: 1;
  top: 0;
  width: 100%;
  height: 100%;
`

const ModalContent = styled.div`
  background-color: ${colors.secondary};
  margin-top: 55px;
  margin-right: 5px;
  padding: 20px;
  border-radius: 15px;
  border: 3px solid ${colors.primary};
  width: 95%;
  max-width: 375px;
  float: right;
  font-size: 1rem;
  @media (min-width: 350px) {
    width: 80%;
  }
`

const ModalLight = ({content}) => {
  return (
    <ModalWrapper>
      <ModalContent>
        {content}
      </ModalContent>
    </ModalWrapper>
  );
}

export default ModalLight;