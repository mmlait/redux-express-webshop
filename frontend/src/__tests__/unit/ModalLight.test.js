import React from 'react';
import { create } from 'react-test-renderer'
import { mount } from 'enzyme';
import ModalLight from '../../components/ModalLight';

let props = {
  content: 'Test content'
}

describe('ModalLight component', () => {
  it('should render', () => {
    let tree = create(<ModalLight />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
  it ('should render with props', () => {
    const wrapper = mount(
      <ModalLight content={props.content} />
    );
    expect(wrapper.props().content).toEqual(props.content);
  })
})