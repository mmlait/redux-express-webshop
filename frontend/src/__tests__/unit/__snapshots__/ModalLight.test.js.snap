// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalLight component should render 1`] = `
<div
  className="sc-bdVaJa iZPlhv"
>
  <div
    className="sc-bwzfXH dQdmMa"
  />
</div>
`;
